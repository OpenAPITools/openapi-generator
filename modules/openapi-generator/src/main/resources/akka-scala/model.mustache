{{>licenseInfo}}
package {{package}}

import {{invokerPackage}}.ApiModel
import org.joda.time.DateTime
import java.util.UUID

{{#models}}
{{#model}}
case class {{classname}} (
  {{#vars}}
  {{#description}}
  /* {{{description}}} */
  {{/description}}
  {{{name}}}: {{^required}}Option[{{/required}}{{^isEnum}}{{datatype}}{{/isEnum}}{{#isEnum}}{{classname}}Enums.{{datatypeWithEnum}}{{/isEnum}}{{^required}}]{{/required}}{{#hasMore}},{{/hasMore}}
  {{/vars}}
) extends ApiModel

{{#hasEnums}}
object {{classname}}Enums {

  {{#vars}}
  {{#isEnum}}
  type {{datatypeWithEnum}} = {{datatypeWithEnum}}.Value
  {{/isEnum}}
  {{/vars}}
  {{#vars}}
  {{#isEnum}}
  object {{datatypeWithEnum}} extends Enumeration {
{{#_enum}}
    val {{#fnEnumEntry}}{{.}}{{/fnEnumEntry}} = Value("{{.}}")
{{/_enum}}
  }

  {{/isEnum}}
  {{/vars}}
}
{{/hasEnums}}
{{/model}}
{{/models}}

