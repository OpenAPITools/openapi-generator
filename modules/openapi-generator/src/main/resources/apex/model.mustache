{{>licenseInfo}}
{{#models}}
{{#model}}
{{#isEnum}}{{>modelEnum}}{{/isEnum}}{{^isEnum}}{{>pojo}}{{/isEnum}}
{{/model}}
{{/models}}
