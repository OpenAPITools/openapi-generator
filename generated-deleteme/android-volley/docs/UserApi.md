# UserApi

All URIs are relative to *https://sim.nonprod.gbcloud.com.au/api*

Method | HTTP request | Description
------------- | ------------- | -------------
[**getCurrentUserEnabledFeatures**](UserApi.md#getCurrentUserEnabledFeatures) | **GET** /v1/users/me/features | Get user features.



## getCurrentUserEnabledFeatures

> GetFeatures getCurrentUserEnabledFeatures()

Get user features.

# Get user features    Gets a list of application features the user is allowed to access  

### Example

```java
// Import classes:
//import org.openapitools.client.api.UserApi;

UserApi apiInstance = new UserApi();
try {
    GetFeatures result = apiInstance.getCurrentUserEnabledFeatures();
    System.out.println(result);
} catch (ApiException e) {
    System.err.println("Exception when calling UserApi#getCurrentUserEnabledFeatures");
    e.printStackTrace();
}
```

### Parameters

This endpoint does not need any parameter.

### Return type

[**GetFeatures**](GetFeatures.md)

### Authorization

No authorization required

### HTTP request headers

- **Content-Type**: Not defined
- **Accept**: application/json

