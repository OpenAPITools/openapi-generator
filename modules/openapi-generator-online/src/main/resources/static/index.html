<!doctype html> <!-- Important: must specify -->
<html>
<head>
  <meta charset="utf-8"> <!-- Important: rapi-doc uses utf8 charecters -->
  <script src="rapidoc-min.js"></script>
</head>
<body>
<rapi-doc spec-url="api-docs"
          theme="dark">
  <img
          slot="logo"
          src="logo.png"
          style="margin:0 -0.3em 0 0.8em;"
  />
</rapi-doc>
</body>
</html>