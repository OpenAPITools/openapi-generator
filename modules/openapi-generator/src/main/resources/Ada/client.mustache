with {{package}}.Clients;
with {{package}}.Models;
with Swagger;
with Util.Http.Clients.Curl;
with Ada.Text_IO;
with Ada.Command_Line;
with Ada.Calendar.Formatting;
with Ada.Exceptions;
procedure {{package}}.Client is

   use Ada.Text_IO;

   procedure Usage;

   Server    : constant Swagger.UString := Swagger.To_UString ("http://localhost:8080/v2");
   Arg_Count : constant Natural := Ada.Command_Line.Argument_Count;
   Arg       : Positive := 1;

   procedure Usage is
   begin
      Put_Line ("Usage: {{projectName}} {params}...");
   end Usage;

begin
   if Arg_Count <= 1 then
      Usage;
      return;
   end if;
   Util.Http.Clients.Curl.Register;
   declare
      Command : constant String := Ada.Command_Line.Argument (Arg);
      Item    : constant String := Ada.Command_Line.Argument (Arg + 1);
      C       : {{package}}.Clients.Client_Type;
   begin
      C.Set_Server (Server);
      Arg := Arg + 2;

   exception
      when E : Constraint_Error =>
         Put_Line ("Constraint error raised: " & Ada.Exceptions.Exception_Message (E));

   end;
end {{package}}.Client;
