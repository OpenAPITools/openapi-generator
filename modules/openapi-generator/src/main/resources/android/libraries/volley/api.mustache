{{>licenseInfo}}
package {{package}};

import {{invokerPackage}}.ApiInvoker;
import {{invokerPackage}}.ApiException;
import {{invokerPackage}}.Pair;

{{#hasModel}}
import {{modelPackage}}.*;
{{/hasModel}}

import java.util.*;

import com.android.volley.Response;
import com.android.volley.VolleyError;

{{#imports}}import {{import}};
{{/imports}}

import org.apache.http.HttpEntity;
import org.apache.http.entity.mime.MultipartEntityBuilder;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.concurrent.ExecutionException;
import java.util.concurrent.TimeoutException;

{{#operations}}
public class {{classname}} {
  String basePath = "{{{basePath}}}";
  ApiInvoker apiInvoker = ApiInvoker.getInstance();

  public void addHeader(String key, String value) {
    getInvoker().addDefaultHeader(key, value);
  }

  public ApiInvoker getInvoker() {
    return apiInvoker;
  }

  public void setBasePath(String basePath) {
    this.basePath = basePath;
  }

  public String getBasePath() {
    return basePath;
  }

  {{#operation}}
  /**
  * {{summary}}
  * {{notes}}
{{#allParams}}   * @param {{paramName}} {{description}}
{{/allParams}}   * @return {{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/returnType}}
  */
  public {{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/returnType}} {{nickname}} ({{#allParams}}{{{dataType}}} {{paramName}}{{^-last}}, {{/-last}}{{/allParams}}) throws TimeoutException, ExecutionException, InterruptedException, ApiException {
    Object postBody = {{#bodyParam}}{{paramName}}{{/bodyParam}}{{^bodyParam}}null{{/bodyParam}};
    {{#allParams}}
    {{#required}}
    // verify the required parameter '{{paramName}}' is set
    if ({{paramName}} == null) {
      VolleyError error = new VolleyError("Missing the required parameter '{{paramName}}' when calling {{nickname}}",
        new ApiException(400, "Missing the required parameter '{{paramName}}' when calling {{nickname}}"));
    }
    {{/required}}
    {{/allParams}}

    // create path and map variables
    String path = "{{{path}}}"{{#pathParams}}.replaceAll("\\{" + "{{baseName}}" + "\\}", apiInvoker.escapeString({{{paramName}}}.toString())){{/pathParams}};

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    // header params
    Map<String, String> headerParams = new HashMap<String, String>();
    // form params
    Map<String, String> formParams = new HashMap<String, String>();
    {{#queryParams}}
    queryParams.addAll(ApiInvoker.parameterToPairs("{{#collectionFormat}}{{{collectionFormat}}}{{/collectionFormat}}", "{{baseName}}", {{paramName}}));
    {{/queryParams}}
    {{#headerParams}}
    headerParams.put("{{baseName}}", ApiInvoker.parameterToString({{paramName}}));
    {{/headerParams}}
    String[] contentTypes = {
      {{#consumes}}
      "{{{mediaType}}}"{{^-last}},{{/-last}}
      {{/consumes}}
    };
    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

    if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder localVarBuilder = MultipartEntityBuilder.create();
      {{#formParams}}
      {{^isFile}}
      if ({{paramName}} != null) {
        localVarBuilder.addTextBody("{{baseName}}", ApiInvoker.parameterToString({{paramName}}), ApiInvoker.TEXT_PLAIN_UTF8);
      }
      {{/isFile}}
      {{#isFile}}
      if ({{paramName}} != null) {
        localVarBuilder.addBinaryBody("{{baseName}}", {{paramName}});
      }
      {{/isFile}}
      {{/formParams}}
      HttpEntity httpEntity = localVarBuilder.build();
      postBody = httpEntity;
    } else {
      // normal form params
      {{#formParams}}
      {{^isFile}}
      formParams.put("{{baseName}}", ApiInvoker.parameterToString({{paramName}}));
      {{/isFile}}
      {{/formParams}}
    }

    String[] authNames = new String[] { {{#authMethods}}"{{name}}"{{^-last}}, {{/-last}}{{/authMethods}} };

    try {
      String localVarResponse = apiInvoker.invokeAPI (basePath, path, "{{httpMethod}}", queryParams, postBody, headerParams, formParams, contentType, authNames);
      if (localVarResponse != null) {
         return {{#returnType}}({{{returnType}}}) ApiInvoker.deserialize(localVarResponse, "{{returnContainer}}", {{returnBaseType}}.class){{/returnType}};
      } else {
         return {{#returnType}}null{{/returnType}};
      }
    } catch (ApiException ex) {
       throw ex;
    } catch (InterruptedException ex) {
       throw ex;
    } catch (ExecutionException ex) {
      if (ex.getCause() instanceof VolleyError) {
        VolleyError volleyError = (VolleyError)ex.getCause();
        if (volleyError.networkResponse != null) {
          throw new ApiException(volleyError.networkResponse.statusCode, volleyError.getMessage());
        }
      }
      throw ex;
    } catch (TimeoutException ex) {
      throw ex;
    }
  }

      /**
   * {{summary}}
   * {{notes}}
{{#allParams}}   * @param {{paramName}} {{description}}{{/allParams}}
  */
  public void {{nickname}} ({{#allParams}}{{{dataType}}} {{paramName}}{{^-last}}, {{/-last}}{{#-last}}, {{/-last}}{{/allParams}}final Response.Listener<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}String{{/returnType}}> responseListener, final Response.ErrorListener errorListener) {
    Object postBody = {{#bodyParam}}{{paramName}}{{/bodyParam}}{{^bodyParam}}null{{/bodyParam}};

    {{#allParams}}
    {{#required}}
    // verify the required parameter '{{paramName}}' is set
    if ({{paramName}} == null) {
      VolleyError error = new VolleyError("Missing the required parameter '{{paramName}}' when calling {{nickname}}",
        new ApiException(400, "Missing the required parameter '{{paramName}}' when calling {{nickname}}"));
    }
    {{/required}}
    {{/allParams}}

    // create path and map variables
    String path = "{{{path}}}".replaceAll("\\{format\\}","json"){{#pathParams}}.replaceAll("\\{" + "{{baseName}}" + "\\}", apiInvoker.escapeString({{{paramName}}}.toString())){{/pathParams}};

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    // header params
    Map<String, String> headerParams = new HashMap<String, String>();
    // form params
    Map<String, String> formParams = new HashMap<String, String>();

    {{#queryParams}}
    queryParams.addAll(ApiInvoker.parameterToPairs("{{#collectionFormat}}{{{collectionFormat}}}{{/collectionFormat}}", "{{baseName}}", {{paramName}}));
    {{/queryParams}}

    {{#headerParams}}
    headerParams.put("{{baseName}}", ApiInvoker.parameterToString({{paramName}}));
    {{/headerParams}}

    String[] contentTypes = {
      {{#consumes}}"{{{mediaType}}}"{{^-last}},{{/-last}}{{/consumes}}
    };
    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

    if (contentType.startsWith("multipart/form-data")) {
      // file uploading
      MultipartEntityBuilder localVarBuilder = MultipartEntityBuilder.create();
      {{#formParams}}{{^isFile}}
      if ({{paramName}} != null) {
        localVarBuilder.addTextBody("{{baseName}}", ApiInvoker.parameterToString({{paramName}}), ApiInvoker.TEXT_PLAIN_UTF8);
      }
      {{/isFile}}{{#isFile}}
      if ({{paramName}} != null) {
        localVarBuilder.addBinaryBody("{{baseName}}", {{paramName}});
      }
      {{/isFile}}{{/formParams}}

      HttpEntity httpEntity = localVarBuilder.build();
      postBody = httpEntity;
    } else {
      // normal form params
      {{#formParams}}{{^isFile}}formParams.put("{{baseName}}", ApiInvoker.parameterToString({{paramName}}));{{/isFile}}
      {{/formParams}}
    }

    String[] authNames = new String[] { {{#authMethods}}"{{name}}"{{^-last}}, {{/-last}}{{/authMethods}} };

    try {
      apiInvoker.invokeAPI(basePath, path, "{{httpMethod}}", queryParams, postBody, headerParams, formParams, contentType, authNames,
        new Response.Listener<String>() {
          @Override
          public void onResponse(String localVarResponse) {
            {{#returnType}}
            try {
              responseListener.onResponse(({{{returnType}}}) ApiInvoker.deserialize(localVarResponse,  "{{returnContainer}}", {{returnBaseType}}.class));
              {{/returnType}}
              {{^returnType}}
              responseListener.onResponse(localVarResponse);
              {{/returnType}}
            {{#returnType}}
            } catch (ApiException exception) {
               errorListener.onErrorResponse(new VolleyError(exception));
            }
            {{/returnType}}
          }
      }, new Response.ErrorListener() {
          @Override
          public void onErrorResponse(VolleyError error) {
            errorListener.onErrorResponse(error);
          }
      });
    } catch (ApiException ex) {
      errorListener.onErrorResponse(new VolleyError(ex));
    }
  }
  {{/operation}}
}
{{/operations}}
