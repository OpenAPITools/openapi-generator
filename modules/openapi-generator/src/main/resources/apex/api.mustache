{{>licenseInfo}}
{{#operations}}
public class {{classname}} {
    {{classPrefix}}Client client;

    public {{classname}}({{classPrefix}}Client client) {
        this.client = client;
    }

    public {{classname}}() {
        this.client = new {{classPrefix}}Client();
    }

    public {{classPrefix}}Client getClient() {
        return this.client;
    }

    {{#operation}}
    /**
     * {{summary}}
     * {{notes}}
     {{#allParams}}
     * @param {{paramName}} {{description}}{{#required}} (required){{/required}}{{^required}} (optional{{#defaultValue}}, default to {{{.}}}{{/defaultValue}}){{/required}}
     {{/allParams}}
     {{#returnType}}
     * @return {{{returnType}}}
     {{/returnType}}
     * @throws OAS.ApiException if fails to make API call
     */
    public {{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/returnType}} {{operationId}}({{#hasParams}}Map<String, Object> params{{/hasParams}}) {
    {{#allParams}}
      {{#required}}
        client.assertNotNull(params.get('{{paramName}}'), '{{paramName}}');
      {{/required}}
    {{/allParams}}
        List<OAS.Param> query = new List<OAS.Param>();
    {{#hasQueryParams}}

        // cast query params to verify their expected type
    {{/hasQueryParams}}
    {{#queryParams}}
      {{#isArray}}
        {{#isCollectionFormatMulti}}
        query.addAll(client.makeParams('{{baseName}}', ({{{dataType}}}) params.get('{{paramName}}')));
        {{/isCollectionFormatMulti}}
        {{^isCollectionFormatMulti}}
        query.addAll(client.makeParam('{{baseName}}', ({{{dataType}}}) params.get('{{paramName}}'), '{{collectionFormat}}'));
        {{/isCollectionFormatMulti}}
      {{/isArray}}
      {{^isArray}}
        query.addAll(client.makeParam('{{baseName}}', ({{{dataType}}}) params.get('{{paramName}}')));
      {{/isArray}}
      {{#-last}}

      {{/-last}}
    {{/queryParams}}
        List<OAS.Param> form = new List<OAS.Param>();
    {{#hasFormParams}}

        // cast form params to verify their expected type
    {{/hasFormParams}}
    {{#formParams}}
      {{#isArray}}
        {{#isCollectionFormatMulti}}
        form.addAll(client.makeParams('{{baseName}}', ({{{dataType}}}) params.get('{{paramName}}')));
        {{/isCollectionFormatMulti}}
        {{^isCollectionFormatMulti}}
        form.addAll(client.makeParam('{{baseName}}', ({{{dataType}}}) params.get('{{paramName}}'), '{{collectionFormat}}'));
        {{/isCollectionFormatMulti}}
      {{/isArray}}
      {{^isArray}}
        form.addAll(client.makeParam('{{baseName}}', ({{{dataType}}}) params.get('{{paramName}}')));
      {{/isArray}}
    {{/formParams}}

        {{#returnType}}return ({{{returnType}}}) {{/returnType}}client.invoke(
            '{{httpMethod}}', '{{path}}',{{#bodyParam}}
            ({{{dataType}}}) params.get('{{paramName}}'){{/bodyParam}}{{^bodyParam}} ''{{/bodyParam}},
            query, form,
            new Map<String, Object>{{#hasPathParams}}{
              {{#pathParams}}
                '{{baseName}}' => ({{{dataType}}}) params.get('{{paramName}}'){{^-last}},{{/-last}}
              {{/pathParams}}
            }{{/hasPathParams}}{{^hasPathParams}}(){{/hasPathParams}},
            new Map<String, Object>{{#hasHeaderParams}}{
              {{#headerParams}}
                '{{baseName}}' => ({{{dataType}}}) params.get('{{paramName}}'){{^-last}},{{/-last}}
              {{/headerParams}}
            }{{/hasHeaderParams}}{{^hasHeaderParams}}(){{/hasHeaderParams}},
          {{#hasProduces}}
            new List<String>{ {{#produces}}'{{mediaType}}'{{^-last}}, {{/-last}}{{/produces}} },
          {{/hasProduces}}
          {{^hasProduces}}
            new List<String>(),
          {{/hasProduces}}
          {{#hasConsumes}}
            new List<String>{ {{#consumes}}'{{mediaType}}'{{^-last}}, {{/-last}}{{/consumes}} },
          {{/hasConsumes}}
          {{^hasConsumes}}
            new List<String>(),
          {{/hasConsumes}}
          {{#hasAuthMethods}}
            new List<String> { {{#authMethods}}'{{name}}'{{^-last}}, {{/-last}}{{/authMethods}} },
          {{/hasAuthMethods}}
          {{^hasAuthMethods}}
            new List<String>(),
          {{/hasAuthMethods}}
          {{#returnType}}
            {{{returnType}}}.class
          {{/returnType}}
          {{^returnType}}
            null
          {{/returnType}}
        );
    }
  {{/operation}}
}
{{/operations}}
