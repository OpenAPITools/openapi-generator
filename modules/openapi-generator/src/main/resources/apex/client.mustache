public class {{classPrefix}}Client extends Swagger.ApiClient {
{{#hasAuthMethods}}
    public {{classPrefix}}Client() {
        basePath = '{{basePath}}';
        calloutName = '{{calloutName}}';
  {{#authMethods}}
    {{#isApiKey}}
      {{#isKeyInQuery}}
        authentications.put('{{name}}', new Swagger.ApiKeyQueryAuth('{{keyParamName}}'));
      {{/isKeyInQuery}}
      {{^isKeyInQuery}}
        authentications.put('{{name}}', new Swagger.ApiKeyHeaderAuth('{{keyParamName}}'));
      {{/isKeyInQuery}}
    {{/isApiKey}}
  {{/authMethods}}
    }
{{/hasAuthMethods}}
{{^hasAuthMethods}}
    public {{classPrefix}}Client() {
        basePath = '{{basePath}}';
        calloutName = '{{calloutName}}';
    }
{{/hasAuthMethods}}
}
