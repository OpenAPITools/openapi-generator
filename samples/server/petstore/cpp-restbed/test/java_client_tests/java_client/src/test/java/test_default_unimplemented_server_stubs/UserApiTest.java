package test_default_unimplemented_server_stubs;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.openapi.example.api.UserApi;
import org.openapi.example.model.User;
import org.springframework.web.reactive.function.client.WebClientResponseException;

import java.util.Arrays;

import static helper.ApiClientFactories.setUpUserApi;
import static org.junit.jupiter.api.Assertions.*;
import static helper.TestingHelper.approveException;

class UserApiTest {

    UserApi apiInstance;

    @BeforeEach
    void setUp() {
        apiInstance = setUpUserApi(1234);
    }

    @Test
    void createUser() {
        User user = new User();
        WebClientResponseException exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.createUser(user).block();
        });

        approveException(exception);
    }

    @Test
    void createUsersWithArrayInput() {
        User user = new User();
        WebClientResponseException exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.createUsersWithArrayInput(Arrays.asList(user)).block();
                });

        approveException(exception);
    }

    @Test
    void createUsersWithListInput() {
        User user = new User();
        WebClientResponseException exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.createUsersWithListInput(Arrays.asList(user)).block();
                });
        approveException(exception);
    }

    @Test
    void loginUser() {
        WebClientResponseException exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.loginUser("my name", "my password").block();
                });

        approveException(exception);
    }

    @Test
    void logoutUser() {
        WebClientResponseException exception = assertThrows(
                WebClientResponseException.class, () -> {
                    apiInstance.logoutUser().block();
                });

        approveException(exception);
    }
}