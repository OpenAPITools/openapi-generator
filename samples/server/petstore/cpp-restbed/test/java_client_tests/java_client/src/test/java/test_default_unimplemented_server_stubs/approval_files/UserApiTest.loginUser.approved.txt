Status code: 501 NOT_IMPLEMENTED
Reason: Not implemented
Response headers: [Connection:"close"]
